File: Python/nyzostrings/nyzostringprivateseed.py
"""
Nyzo String for private Seed
"""

from nyzostrings.nyzostring import NyzoString


__version__ = "0.0.1"


class NyzoStringPrivateSeed(NyzoString):

    def __init__(self, identifier: bytes) -> None:
        super().__init__('key_', identifier)

    def get_seed(self) -> bytes:
        return self.bytes_content

    @classmethod
    def from_hex(cls, hex_string: str) -> "NyzoStringPrivateSeed":
        filtered_string = hex_string.replace('-', '')[:64]
        return NyzoStringPrivateSeed(bytes.fromhex(filtered_string))

File: JavaScript/src/NyzoStringPrivateSeed.js
// ref: https://github.com/n-y-z-o/nyzoVerifier/blob/master/src/main/java/co/nyzo/verifier/nyzoString/NyzoStringPublicIdentifier.java

const { NyzoString } = require('./NyzoString')


class NyzoStringPrivateSeed extends NyzoString {

  constructor(identifier) {
    super('key_', identifier)
  }

  getSeed() {
        return this.bytes
  }

  static fromHex(hexString) {
    const filteredString = hexString.split('-').join('').slice(0, 64)
    const buffer = Buffer.from(filteredString, 'hex')
    return new NyzoStringPrivateSeed(buffer)
  }

}


module.exports = {
    version: "0.0.1",
    NyzoStringPrivateSeed
}
